File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
		String bin = Integer.toString(n, k);
		String[] arr = bin.split("0");
		for(int i = 0; i < arr.length; i ++) {
			if(isPrimeNumber(arr[i])) {
				answer++;
			}
		}
        return answer;
    }
    public boolean isPrimeNumber(String num) {
		if(num.isEmpty()) return false;
		Long number = Long.parseLong(num);
		if(number == 1) return false;
		for(long i = 2; i <= Math.sqrt(number); i++) {
			if(number % i == 0) return false;
		}
		return true;
	}
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
const isPrime = (num) => {
    if (isNaN(num) || num < 2) return false;
    for (let k = 2; k <= num / k; k++){
        if (num % k === 0) return false;
    }
    return true;
}

function solution(n, k) {
    let answer = 0;
    nums = n.toString(k).split('0');
    return nums.filter(num => isPrime(num)).length;
}